resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerUtils.scala

```diff
@@ -451,15 +468,22 @@ trait MesosSchedulerUtils extends Logging {
   }
 
   /** Creates a mesos resource for a specific port number. */
-  private def createResourcesFromPorts(portsAndRoles: List[(Long, String)]) : List[Resource] = {
-    portsAndRoles.flatMap{ case (port, role) =>
-      createMesosPortResource(List((port, port)), Some(role))}
+  private def createResourcesFromPorts(
+       portsAndResourcesInfo: List[(Long, (String, AllocationInfo, Option[ReservationInfo]))])
```

Comment:
`(String, AllocationInfo, Option[ReservationInfo])` is kind of an awkward type. A case class to wrap that info might help here.

Reply:
yeah makes sense.